### PR TITLE
Draining implemented

### LINKED ISSUE #1721: Draining timeout in launch config
Provide an option in group launch config to store draining timeout. Following can be the new launch config:
```
{
   "launchConfiguration":{
      "args":{
         "loadBalancers":[{"port": 80, "loadBalancerId": 237935}],
         "server":{
            "name":"autoscale_server",
            "imageRef":"7cf5ffc3-7b20-46fd-98e4-fefa9908d7e8"         
         },
        "draining_timeout": 30
      },
      "type":"launch_server"
   },
  ...
}
```
- [x] Implement in otter
- [x] Integration test
- [ ] Docs

### PR DESCRIPTION
**Pull request by manishtomar**
Taking draining timeout from launch config and integration test for draining. With this draining is complete. Needs https://github.com/rackerlabs/mimic/pull/428 merged before running integration tests.

Implements 1st and 2nd task of #1721.

### CODE REVIEW

File: otter/integration/tests/test_lbsh.py [hunk 1]
```diff
@@ -599,3 +602,54 @@ def test_changing_disowned_server_is_not_converged_2(self):
                 timeout=timeout_default
             )
         ])
+
+    @inlineCallbacks
+    def test_draining(self):
```
Comment (radix) [inline otter/integration/tests/test_lbsh.py]: Do you expect this test to pass against production, given it has those strict timing requirements?  Maybe this test should have a `@skip_if(not_mimic, "Requires mimic for timing")` or something?

**Comment by radix** [comment]
Just one minor point/question. LGTM!

File: otter/integration/tests/test_lbsh.py (hunk 1 above)
Comment (manishtomar) [inline otter/integration/tests/test_lbsh.py]: I've tested this against prod. I am _hoping_ that `30 + 5 + convergence_interval * 2` should make it work in prod as long as `convergence_interval` matches one in prod.

Comment (radix) [inline otter/integration/tests/test_lbsh.py]: okay, cool! feel free to merge then :)

**Pull request merged**